internal/kubernetes/operator/config_exporter.go

```diff
@@ -395,3 +402,33 @@ func (e *ConfigExporter) exportAtlasStreamProcessing(projectName string) ([]runt
 
 	return result, nil
 }
+
+func (e *ConfigExporter) exportAtlasFederatedAuth(projectName string) ([]runtime.Object, error) {
+	result := make([]runtime.Object, 0)
+	// Gets the FederationAuthSetting
+	federatedAuthentificationSetting, err := e.dataProvider.FederationSetting(&admin.GetFederationSettingsApiParams{OrgId: e.orgID})
+	if err != nil {
+		return nil, err
+	}
+	// Does not have an IdenityProvider set then no need to generate
+	if !federatedAuthentificationSetting.HasIdentityProviderStatus() || federatedAuthentificationSetting.GetIdentityProviderStatus() == "INACTIVE" {
+		return nil, err
+	}
+	// Does have an IdentityProvider and then we can generate the config
+	federatedAuthentification, err := federatedauthentication.BuildAtlasFederatedAuth(&federatedauthentication.AtlasFederatedAuthBuildRequest{
+		IncludeSecret:                e.includeSecretsData,
+		IdentityProviderLister:       e.dataProvider,
+		ConnectedOrgConfigsDescriber: e.dataProvider,
+		IdentityProviderDescriber:    e.dataProvider,
+		ProjectStore:                 e.dataProvider,
+		ProjectID:                    e.projectID,
+		OrgID:                        e.orgID,
+		TargetNamespace:              e.targetNamespace,
+		Version:                      e.operatorVersion,
+		Dictionary:                   e.dictionaryForAtlasNames,
+		ProjectName:                  projectName,
+		FederatedSettings:            federatedAuthentificationSetting,
+	})
+	result = append(result, federatedAuthentification)
+	return result, err
```

Comment:
I would also add context here to the error and split the error codepath completely:
```suggestion
	if err != nil {
	    return nil, fmt.Errorf("failed to export federated authentication: %w", err)
	} 
	return append(result, federatedAuthentification)
```